### ISSUE TITLE
Support templates

### ISSUE THREAD

**Issue by davedelong** [subdigital/nsdateformatter.com#11]
`NSDateFormatter` has some pretty awesome API to turn templates into locale-specific date format strings.

IE, `j:m` will turn into `h:mm`, `HH:mm`, etc depending on your locale. It'd be awesome to see this exposed on the site.

**Comment by subdigital** [comment]
ooh, that's pretty neat.  I didn't know this.

**Comment by subdigital** [comment]
Coming back around to this, I haven't ever used/needed this before so I'm not sure if this is something that _needs_ to be reflected on the site. Happy to review a PR, or even if some reference examples are provided, I can turn them into a new tabbed section.

**Comment by jablair** [comment]
So, I just used the "j" formatter this week and came to see if there was an issue for it.

We got a ping for our localization testers that we weren't properly respecting non-US standard for time display.

Here's an example

```swift
let usLocale = Locale(identifier: "en_US")
let deLocale = Locale(identifier: "de_DE")

let usFormat = DateFormatter.dateFormat(fromTemplate: "MM/dd/yyyy j:mm", options: 0, locale: usLocale)
// usFormat returned as "MM/dd/yyyy, h:mm a"

let deFormat = DateFormatter.dateFormat(fromTemplate: "MM/dd/yyyy j:mm", options: 0, locale: deLocale) 
// deFormat returned as "dd.MM.yyyy, HH:mm"
```

So, you get AM/PM formatting in the US locale and 24 hour formatting in the Germany locale.

More directly, you can do this:

```swift
// System region set to Germany
let dateFormatter = DateFormatter()
dateFormatter.setLocalizedDateFormatFromTemplate("MM/dd/yyyy j:mm") 
dateFormatter.string(from: Date())
// String = "25.10.2019, 17:14"
```

I learned the hard way that just calling `setLocalizedDateFormatFromTemplate("MM/dd/yyyy hh:mm a")` isn't smart enough to translate the time into an actual localised time format.

**Comment by subdigital** [comment]
Addressed by #43
